File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.py
def solution(n):
    answer = ''
    num = ['4', '1', '2'] # 3진법의 0 -> 124수의 4, 1 -> 1, 2 - > 2
    
    while n > 0:
        n, rest = n // 3, n % 3
        answer = num[rest] + answer
        if rest == 0:
            n -= 1
            
    return answer

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
class Solution {
    public String solution(int n) {
        String answer = "";
        //3진법의 0 -> 124수의 4, 1 -> 1, 2 - > 2
        String[] num = new String[] {"4", "1", "2"};
        
        while (n > 0) {
            int rest = n % 3;
            answer = num[rest] + answer;
            n /= 3;
            if (rest == 0)
                n--;
        }
        return answer;
    }
}